libraries/AP_HAL/Device.h

```diff
@@ -36,7 +36,8 @@ class AP_HAL::Device {
         SPEED_LOW,
     };
 
-    typedef void PeriodicHandle;
+    FUNCTOR_TYPEDEF(PeriodicCb, bool);
+    typedef void* PeriodicHandle;
```

Comment:
Any specific reason why this needs to be an opaque pointer type? If this was an object, `adjust_periodic_callback()` could belong to it instead of `AP_HAL::Device`, and `unregister_callback()` probably wouldn't be necessary (as it already seems that you're aware of that, by the commit message).